lib/syskit/task_context.rb

```diff
@@ -959,6 +983,14 @@ def setup_failed!(exception)
 
             state_reader.resume if state_reader.respond_to?(:resume)
 
+            unless needs_setup?
```

Comment:
**Here** is why `needs_setup?` is plain wrong. Why does `setup` (which relates to configuration) affects start ?

Just use read_only directly

Reply:
changed